code-challenges.test.js

```diff
@@ -21,39 +21,135 @@ const numbersArray2 = [24, 27, 30, 33, 36]
 const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
 // Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
+    expect(capitalLetters(["streetlamp", "potato", "teeth", "conclusion", "nephew"])).toEqual(["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"])
+    expect(capitalLetters(["temperature", "database", "chopsticks", "mango"])).toEqual(["Temperature", "Database", "Chopsticks", "Mango"])
+  })
+})
+// output ---> capitalLetters is not defined
 
 // b) Create the function that makes the test pass.
 
+// psuedocode:
+// input: an array of various words
+// using the built in method .map, iterate through the array and using charAt pinpoint the first character of each word and .toUpperCase to make each of them capitalized
+// output: will return an array with all of the first letters of the words capitalized 
+
+const capitalLetters = (array) => {
+    return array.map((value) => {
+        return value.charAt(0).toUpperCase() + value.slice(1)
+    })
+}
+
+
+
 // --------------------3) Create a function that takes in a string and logs the index of the first vowel.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const vowelTester1 = "learn"
+describe("vowelExtractor", () => {
+    const vowelTester1 = "learn"
 // Expected output: 1
 const vowelTester2 = "academy"
 // Expected output: 0
 const vowelTester3 = "challenges"
 // Expected output: 2
+it("returns the index of the first vowel", () => {
+    expect(vowelExtractor(vowelTester1)).toEqual(1)
+    expect(vowelExtractor(vowelTester2)).toEqual(0)
+    expect(vowelExtractor(vowelTester3)).toEqual(2)
+  })
+})
+
+// Pseudocode: 
+// input: a string
+// create a function called vowelExtractor that takes in a string as a parameter. Create the vowelsArray, split the input string into an array. Use forEach to iterate through each character(item) to use in the conditional. In the conditional its checking for the index using the built in method indexOf to find the vowel if the index of it does not equal -1, meaning it does exist within the string and meets the other condition of being undefined since vowelsIndex is not assigned until after the conditions are both met. When both conditions are met vowelsIndex is assigned to the index of the vowel appearing in the given input
+// output: will return the index position of the first vowel in the given string
 
 // b) Create the function that makes the test pass.
+
+
+const vowelsArray = ["a", "e", "i", "o", "u", "y"]
+
+const vowelTester1 = "learn"
+// Expected output: 1
+const vowelTester2 = "academy"
+// Expected output: 0
+const vowelTester3 = "challenges"
+// Expected output: 2
+
+function vowelExtractor(string) {
+    let vowelIndex
+    string.split("").forEach(item => {
+        if(vowelsArray.indexOf(item) != -1 && vowelIndex === undefined) {
+            vowelIndex = string.indexOf(item) 
+         }
+    });
+    return vowelIndex
+}
```

Comment:
Great job with the pseudocode explaining what is going on! Can you refactor and use arrow function?